azure/Guidelines.md

```diff
@@ -477,6 +477,210 @@ Overall, this is a very brittle design that leads to a poor developer experience
+not                  | Logical negation      | not price le 3.5
+Grouping Operators   |                       |
+( )                  | Precedence grouping   | (priority eq 1 or city eq 'Redmond') and price gt 100
+
+:white_check_mark: **DO** respond with an error message as defined in the [Unsupported Requests](??) section if a client includes an operator in a _filter_ expression that is not supported by the operation.
+
+:white_check_mark: **DO** use the following operator precedence for supported operators when evaluating _filter_ expressions. Operators are listed by category in order of precedence from highest to lowest. Operators in the same category have equal precedence:
+
+| Group           | Operator | Description           |
+|:----------------|:---------|:----------------------|
+| Grouping        | ( )      | Precedence grouping   |
+| Unary           | not      | Logical Negation      |
+| Relational      | gt       | Greater Than          |
+|                 | ge       | Greater than or Equal |
+|                 | lt       | Less Than             |
+|                 | le       | Less than or Equal    |
+| Equality        | eq       | Equal                 |
+|                 | ne       | Not Equal             |
+| Conditional AND | and      | Logical And           |
+| Conditional OR  | or       | Logical Or            |
+
+##### Operator examples
+The following examples illustrate the use and semantics of each of the logical operators.
+
+Example: all products with a name equal to 'Milk'
+
+```http
+GET https://api.contoso.com/v1.0/products?filter=name eq 'Milk'
+```
+
+Example: all products with a name not equal to 'Milk'
+
+```http
+GET https://api.contoso.com/v1.0/products?filter=name ne 'Milk'
+```
+
+Example: all products with the name 'Milk' that also have a price less than 2.55:
+
+```http
+GET https://api.contoso.com/v1.0/products?filter=name eq 'Milk' and price lt 2.55
+```
+
+Example: all products that either have the name 'Milk' or have a price less than 2.55:
+
+```http
+GET https://api.contoso.com/v1.0/products?filter=name eq 'Milk' or price lt 2.55
+```
+
+Example: all products that have the name 'Milk' or 'Eggs' and have a price less than 2.55:
+
+```http
+GET https://api.contoso.com/v1.0/products?filter=(name eq 'Milk' or name eq 'Eggs') and price lt 2.55
+```
+
+#### orderby
+
+:heavy_check_mark: **YOU MAY** support sorting of the results of a list operation with the _orderby_ query parameter.
+
+The value of the _orderby_ parameter is a comma-separated list of expressions used to sort the items.
+A special case of such an expression is a property path terminating on a primitive property.
+
+Each expression in the _orderby_ parameter value may include the suffix "asc" for ascending or "desc" for descending, separated from the expression by one or more spaces.
+
+:white_check_mark: **DO** sort the collection in ascending order on an expression if "asc" or "desc" is not specified.
```

Comment:
Should we prescribe lexicographic ordering?

Reply:
A little bit below we say "use the inherent sort order for the type of the property".  We _could_ make that more specific.  Should we?